sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecordCoder.java

```diff
@@ -50,37 +49,22 @@ public KafkaRecordCoder(Coder<K> keyCoder, Coder<V> valueCoder) {
   }
 
   @Override
-  public void encode(KafkaRecord<K, V> value, OutputStream outStream)
-      throws CoderException, IOException {
-    encode(value, outStream, Context.NESTED);
+  public void encode(KafkaRecord<K, V> value, OutputStream outStream) throws IOException {
+    stringCoder.encode(value.getTopic(), outStream);
```

Comment:
Why this change? Is it also a semantic no-op? (i.e. update-compatible)

Reply:
Looks like context is not used anymore and API is deprecated. Must be a no-op. May be SDK runtime has stopped differentiating 'outer' and 'nested'. I noticed deprecation warnings since couple of weeks back and thought it might be safer to try in 2.4. Deprecation messages in JavaDoc for `Context` does not talk about why it is removed and does not warn about any unintended consequences. Important coders `KvCoder` seem to ignore it as well..